modules/react/button/lib/TertiaryButton.tsx

```diff
@@ -97,65 +99,89 @@ const getTertiaryButtonColors = (
+        icon: children ? themePrimary.dark : colors.blackPepper500,
         label: themePrimary.dark,
       },
       active: {
         background: colors.soap300,
-        icon: themePrimary.dark,
+        icon: children ? themePrimary.dark : colors.blackPepper500,
         label: themePrimary.dark,
       },
       focus: {
-        icon: themePrimary.dark,
-        label: themePrimary.dark,
-        focusRing: focusRing({}, theme),
+        icon: children ? themePrimary.main : colors.blackPepper500,
+        label: themePrimary.main,
+        focusRing: focusRing({innerColor: colors.blueberry400}, theme),
       },
       disabled: {
-        icon: themePrimary.main,
+        icon: children ? themePrimary.main : colors.blackPepper400,
         label: themePrimary.main,
       },
     };
   }
 };
 
+const StyledButtonLabel = styled(BaseButton.Label)({
+  textDecoration: 'underline',
+});
+
+const StyledTertiaryButtonContainer = styled(BaseButton)<
+  StyledType & Pick<TertiaryButtonProps, 'allCaps'>
+>(
+  {
+    border: '0',
+  },
+  ({allCaps}) => ({
+    textTransform: allCaps ? 'uppercase' : undefined,
+  })
+);
+
 const getPaddingStyles = (
   icon: CanvasSystemIcon | undefined,
-  iconPosition: 'left' | 'right'
+  iconPosition: 'start' | 'end',
+  children: React.ReactNode
 ): string => {
-  // if there is an icon on the left, add 4px extra padding to the right for visual balance
-  if (icon && iconPosition === 'left') {
-    return `0 ${space.xs} 0 ${space.xxs}`;
-  }
-  // if there is an icon on the right, add 4px extra padding to the left for visual balance
-  if (icon && iconPosition === 'right') {
-    return `0 ${space.xxs} 0 ${space.xs}`;
+  if (children) {
+    // if there is an icon on the left, add 4px extra padding to the right for visual balance
+    if (icon && iconPosition === 'start') {
+      return `0 ${space.xs} 0 ${space.xxs}`;
+    }
+    // if there is an icon on the right, add 4px extra padding to the left for visual balance
+    if (icon && iconPosition === 'end') {
+      return `0 ${space.xxs} 0 ${space.xs}`;
+    }
+    // if there is no icon, return the default padding
+    return `0 ${space.xxs}`;
+  } else {
+    return '0';
   }
-  // if there is no icon, return the default padding
-  return `0 ${space.xxs}`;
 };
 
-// All disabled buttons are set to 40% opacity.
-// This will eventually live in the ButtonContainer styles, but for now we're scoping it to Primary, Secondary, and Tertiary buttons.
-const disabledButtonOpacity = {
-  '&:disabled, &:disabled:active': {
-    opacity: 0.4,
-  },
+const getMinWidthStyles = (children: React.ReactNode, size: ButtonSizes): string => {
+  switch (size) {
+    case 'large':
+      return children ? 'auto' : '48px';
+    case 'medium':
+      return children ? 'auto' : space.xl;
+    case 'small':
+      return children ? 'auto' : space.l;
+    case 'extraSmall':
+      return children ? 'auto' : space.m;
+    default:
+      return children ? 'auto' : space.xl;
+  }
```

Comment:
Can we move out repeated conditions from switch? It takes more performance time than reassigning a variable.

<img width="468" alt="Screen Shot 2022-03-31 at 10 13 53 AM" src="https://user-images.githubusercontent.com/48605821/161101660-f8582b99-84ea-4c08-b37b-6ed0dbba2b3f.png">

<img width="496" alt="Screen Shot 2022-03-31 at 10 18 45 AM" src="https://user-images.githubusercontent.com/48605821/161102703-bef61e56-4737-497d-a3d6-5ed4c65c5c56.png">

<img width="388" alt="Screen Shot 2022-03-31 at 10 18 28 AM" src="https://user-images.githubusercontent.com/48605821/161102618-d676a499-91e7-4be7-a13b-3b9254b57c72.png">

Reply:
Link to replit: https://replit.com/@RayRedGoose/Nodejs-Beta#index.js

Reply:
But my favorite is:

```
const thirdFunction = (size, children, space = {}) => {
  const {xl: any, l: small, m: extraSmall} = space;
  
  const sizeNumbers = {
    large: '48px',
    small,
    extraSmall,
    any
  };
  
  return children 
    ? 'auto' 
    : sizeNumbers[size] || sizeNumbers.any 
}
```